### PR TITLE
Edit dest_validation process

### PR DESCRIPTION
**Pull request by nicholas-david**
This pull request serves to edit and work on the destination validation process.

### CODE REVIEW

**Comment by nicholas-david** [comment]
@DRosasMoctezuma 
I just finished a preliminary pass through the destination validation script. Let me know what you think.

**Comment by nicholas-david** [comment]
@gboeing Hey Geoff,
I finished up cleaning the destination validation script. Let me know if you think it is ready to add to. Thanks

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,8 @@
 *.graphml
 process/data/input/*.csv
 validation/edge/data
-validation/poi/data
+validation/destination/data
```
Comment (gboeing) [inline .gitignore:7]: will we still need this line in the ignore file, given that the data have been moved to `/validation/data/` now?

File: validation/destination/destination_validation.py [hunk 2]
```diff
@@ -1,67 +1,186 @@
...
+    -------
+    destinations_count : value
+    	the value of the number of destinations contained in the gdf
+    """
+    destinations_count = len(gdf_destinations)
+    return destinations_count
+
+def calculate_intersect(a, b, dist):
+    """
+    Calculate XXX.
```
Comment (gboeing) [inline validation/destination/destination_validation.py]: looks like a placeholder "xxx" was left in this docstring

File: validation/destination/destination_validation.py [hunk 3]
```diff
@@ -1,67 +1,186 @@
...
+        print(ox.ts(), 'projected official destinations')
+    if gdf_osm_destinations.crs != crs:
+        gdf_osm_destinations = gdf_osm_destinations.to_crs(crs)
+        print(ox.ts(), 'projected osm destinations')
+    
+    # double-check everything has same CRS, then return
+    assert gdf_study_area.crs == gdf_official_destinations.crs == gdf_osm_destinations.crs
+    return study_area, gdf_official_destinations, gdf_osm_destinations
+
+def total_destination_count(gdf_destinations):
```
Comment (gboeing) [inline validation/destination/destination_validation.py]: we can simplify this script by deleting this function, and whenever it is called elsewhere just replace that call with `len(gdf_whatever)` instead

File: validation/destination/destination_validation.py [hunk 4]
```diff
@@ -1,67 +1,186 @@
...
+
+    # find the portion of each's buffered geometry that intersects with the other's buffered geometry
+    a_buff_overlap_count = a_buff_unary.intersection(b_buff_unary)
+    b_buff_overlap_count = b_buff_unary.intersection(a_buff_unary)
+
+    return intersections_count, percent_intersections, a_buff_overlap_count, b_buff_overlap_count
+
+def min_distance(a, b):
+    """
+    Load the city destinations and study boundary.
```
Comment (gboeing) [inline validation/destination/destination_validation.py]: I think this function description line is incorrect?

File: validation/destination/destination_validation.py [hunk 5]
```diff
@@ -1,67 +1,186 @@
...
+    Parameters
+    ----------
+    a : geopandas.GeoDataFrame
+        the osm or offical destinations
+    b : geopandas.GeoDataFrame
+        the osm or offical destinations
+
+    Returns
+    -------
+    XXX NEEDS TO BE ADDED XXX
```
Comment (gboeing) [inline validation/destination/destination_validation.py]: Looks like the function isn't returning anything right now. Should it be?

File: validation/destination/destination_validation.py [hunk 6]
```diff
@@ -1,67 +1,186 @@
...
+    # calculate total street length and edge count in each dataset, then add to indicators
+    osm_dest_count= total_destination_count(gdf_osm_destinations)
+    official_dest_count = total_destination_count(gdf_official_destinations)
+    indicators[city]['osm_dest_count'] = osm_dest_count
+    indicators[city]['official_dest_count'] = official_dest_count
+    print(ox.ts(), 'calculated destination counts')
+
+    # calculate the % overlaps of areas and lengths between osm and official streets with different buffer distances
+    for dist in edge_buffer_dists:
+		osm_intersections_count, official_intersections_count, osm_percent_intersections, official_percent_intersections, osm_buff_overlap_count, official_buff_overlap_count = calculate_intersect(gdf_osm_destinations, gdf_official_destinations, dist)
```
Comment (gboeing) [inline validation/destination/destination_validation.py]: looks like misaligned indentation in this for loop?

**Review by gboeing — CHANGES_REQUESTED** [review]
Thanks! Just a few comments, inline.

File: .gitignore (hunk 1 above)
Comment (nicholas-david) [reply .gitignore:7]: No, I am keeping it while we transition though.

**Comment by nicholas-david** [comment]
@gboeing Hey Geoff, when I try to run this, I keep getting the following error:

Screen Shot 2020-07-03 at 3 48 45 PM

The same thing happens when I try to run the edge validation code.

Do you know what I should do?

File: validation/configuration/olomouc.json [hunk 7]
```diff
@@ -1,8 +1,7 @@
...
     "study_region"             : "olomouc",
-    "osm_graphml_path"         : "data/Olomouc/olomouc_cz_2019_10000m_all_osm_20190902.graphml",
-    "osm_buffer_gpkg_path"     : "data/Olomouc/olomouc_cz_2019_1600m_buffer.gpkg",
-    "official_streets_shp_path": "data/Olomouc/olomouc_edges.shp",
-    "official_dests_filepath"  : "data/Olomouc/Olomouc_supermarkets_2018.shp",
-    "dests_column_name"        : "fresh_food_market"
+    "osm_graphml_path"         : "../data/olomouc/olomouc_cz_2019_10000m_all_osm_20190902.graphml",
+    "osm_buffer_gpkg_path"     : "../data/olomouc/olomouc_cz_2019_1600m_buffer.gpkg",
+    "official_streets_shp_path": "../data/olomouc/Olomouc_Street_Network/",
+    "official_dests_filepath"  : "../data/olomouc/olomouc_supermarkets_2018/",
```
Comment (gboeing) [inline validation/configuration/olomouc.json]: @nicholas-david the problem is probably this stray trailing comma at the end of this line

Comment (gboeing) [reply validation/configuration/olomouc.json]: > Do you know what I should do?

See inline comment on code here.

**Comment by nicholas-david** [comment]
@gboeing The destination script now runs! I think we will need to rethink some of the code, however, because we are not getting indicators from the uniary unions. The uniary unions are giving us multipolygons, no the counts of how many destinations are overlapping.

**Comment by gboeing** [comment]
Thanks, I'll merge now take a longer look next week.

**Pull request merged**